src/gtc/numpy/npir_codegen.py

```diff
@@ -72,6 +72,8 @@ def shim_key(self, key):
                         new_args.append(idx + offset)
             if not isinstance(new_args[2], (numbers.Integral, slice)):
                 assert isinstance(new_args[0], slice) and isinstance(new_args[1], slice)
+                if np.max(new_args[2]) >= self.field_view.shape[2] or np.min(new_args[2]) < 0:
```

Comment:
I think that should be a function with a descriptive name, or at least a comment that describes this special case.